src/geneva/Go2.cpp

```diff
@@ -891,6 +893,18 @@ void Go2::parseCommandLine(
 			);
 		}
 
+		if ( vm.count("ioc") && m_ioc != 0 ){
```

Comment:
This does not look like the right place to register the consumer.

Reply:
well here i did not found a easy way to implement that, may be you could help?